### DISCUSSION TITLE
Is there a list of S3 compatible etcd services for backup?

### DISCUSSION THREAD

**Discussion by husarcik** [k3s-io/k3s#8140] [Q&A]
I've noticed S3 etcd backup fails with Cloudflare R2 as it cannot detect buckets during cron job but works fine with manual save. Is there a list of S3 compatible services that work besides Amazon?

**Comment by brandond** [comment]
We use the minio golang S3 client libraries; anything that provides a S3 compatible API should work. What specifically do you mean by "cannot detect buckets"?

**Comment by maggie44** [comment]
I am seeing the following error when doing a backup:

```
ERRO[0001] Error retrieving S3 snapshots for reconciliation: ListObjectsV2 search parameter metadata not implemented 
```

Looks like the search parameters used are not implemented with Cloudflare R2. 
https://developers.cloudflare.com/r2/reference/changelog/#2022-07-01

Backups have been working though, both manually and via daily automated backups, this is the only issue I have encountered the last few months. 

Not sure whether this info is ok here, or a bug, or a feature request for Cloudflare R2 support?
